linkerd/stack/tracing/src/lib.rs

```diff
@@ -25,9 +24,8 @@ pub struct NewInstrument<G, N> {
 }
 
 /// Instruments a service produced by `NewInstrument`.
-#[pin_project]
 #[derive(Debug)]
-pub struct Instrument<T, G, S> {
+pub struct Instrument<T, G: GetSpan<T>, S> {
```

Comment:
any particular reason to always require this bound?

Reply:
We need it on the `Drop` and all bound on the `Drop` must also be on the struct definition.

Reply:
yup, i just realized that!